Document more exceptions (#3163)
Azure/azure-iot-sdk-csharp: SDK v2 migration guide.md

| `MethodRequest` | `DirectMethodRequest` | Use full name of the operation type.⁴ |
| `MethodResponse` | `DirectMethodResponse` | See⁴ |
| `IotHubException` | `IotHubClientException` | Specify the exception is for Hub device and module client only. |

> | `DeviceAuthenticationWithTokenRefresh` and `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithSharedAccessKeyRefresh` | More descriptive naming and reduce duplication.⁵ |

Reviewer: ```suggestion
| `DeviceAuthenticationWithTokenRefresh` and `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithTokenRefresh` | More descriptive naming and reduce duplication.⁵ |
```

Author: Sloppy, sorry.

> | `DeviceAuthenticationWithTokenRefresh.` and `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithSharedAccessKeyRefresh` | More descriptive naming and reduce duplication.⁵ |

Reviewer: ```suggestion
| `DeviceAuthenticationWithTokenRefresh` and `ModuleAuthenticationWithTokenRefresh` | `ClientAuthenticationWithSharedAccessKeyRefresh` | More descriptive naming and reduce duplication.⁵ |
```

| `DeviceAuthenticationWithToken` and `ModuleAuthenticationWithToken` | `ClientAuthenticationWithSharedAccessSignature` | See⁵ |

> | `DeviceAuthenticationWithSakRefresh` and `ModuleAuthenticationWithSakRefresh` | `ClientAuthenticationWithTokenRefresh` | See⁵ |

Reviewer: I doubt this, should we exchange the newer API names between L200 and L202?

Author: You're right. Fixed.

> | `DeviceAuthenticationWithTokenRefresh.SafeCreateNewToken(...)` and `ModuleAuthenticationWithTokenRefresh.SafeCreateNewToken(...)` | `ClientAuthenticationWithSharedAccessKeyRefresh.SafeCreateNewTokenAsync(...)` | Async suffix for async methods. |

Reviewer: Is it supposed to be like this?

```suggestion
| `DeviceAuthenticationWithSakRefresh.SafeCreateNewToken(...)` and `ModuleAuthenticationWithSakRefresh.SafeCreateNewToken(...)` | `ClientAuthenticationWithSharedAccessKeyRefresh.SafeCreateNewTokenAsync(...)` | Async suffix for async methods. |
```

Author: Sort of. Fixed.
